### PR TITLE
DOC: Fixes flake8 issues in whatsnew v0.13.* #24256

### LINKED ISSUE #24256: DOC: Fix flake8 issues in doc/source/whatsnew/v0.13.*.rst
xref: #24177 

We didn't start validating the format of PEP8 and other code standards in the documentation examples until recently. We still have some files with errors, that we need to skip, and that we should fix, so we can start validating them.

**Two steps are required to replicate the issue**

1. Edit `setup.cfg` in the pandas home, and in the flake8-rst section, remove from the `exclude` list the file `doc/source/whatsnew/v0.13.0.rst` & `doc/source/whatsnew/v0.13.1.rst` 

2. Remove the import block ( `from pandas import *`) from the lines (8-11) of the file `v0.13.*.rst`. 

After that, running the next command will report the errors in the file (note that syntax error usually prevent to validate other errors, and the list of errors to fix can become much longer when the syntax error is fixed (please make sure that you are using `flake8-rst` version 0.7.0 or higher):

```
$ flake8-rst doc/source/whatsnew/v0.13.*
doc/source/whatsnew/v0.13.0.rst:107:47: E261 at least two spaces before inline comment
doc/source/whatsnew/v0.13.0.rst:124:10: E111 indentation is not a multiple of four
doc/source/whatsnew/v0.13.0.rst:124:13: E999 SyntaxError: invalid syntax
doc/source/whatsnew/v0.13.0.rst:149:26: E231 missing whitespace after ':'
doc/source/whatsnew/v0.13.0.rst:149:33: E231 missing whitespace after ','
...
doc/source/whatsnew/v0.13.1.rst:33:28: E251 unexpected spaces around keyword / parameter equals
doc/source/whatsnew/v0.13.1.rst:33:30: E251 unexpected spaces around keyword / parameter equals
doc/source/whatsnew/v0.13.1.rst:33:46: E231 missing whitespace after ','
doc/source/whatsnew/v0.13.1.rst:33:52: E231 missing whitespace after ','
doc/source/whatsnew/v0.13.1.rst:33:58: E231 missing whitespace after ','
...
```

Once all the errors are addressed, please open a pull request with the fixes in the file, and removing the file from setup.cfg. If you need to do something that feels wrong to fix an error, please ask in a comment to this issue. Please avoid other unrelated changes, which can be addressed in a separate pull request.
cc: @datapythonista

### PR DESCRIPTION
**Pull request by Seuss27**
- [ ] closes #24256

### CODE REVIEW

**Comment by Seuss27** [comment]
I am not sure if there is anything different I can do with the merge conflict in setup.cfg.  Please advise.

File: doc/source/whatsnew/v0.13.1.rst [hunk 1]
```diff
@@ -89,10 +86,13 @@ Output Formatting Enhancements
...
-  Previously output might look like:
-
-  .. code-block:: python
+  Previously output might look like:::
 
+      >>> df = pd.DataFrame([pd.Timestamp('20010101'),
+                            pd.Timestamp('20040601')], columns=['age'])
+      >>> df['today'] = pd.Timestamp('20130419')
+      >>> df['diff'] = df['today'] - df['age']
+      >>> df
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.13.1.rst]: I'm happy with that, but I think there are `.. code-block:: text` or something like that, which would show in a block but won't raise the error.

File: setup.cfg [hunk 2]
```diff
@@ -138,7 +116,7 @@ known_post_core=pandas.tseries,pandas.io,pandas.plotting
 sections=FUTURE,STDLIB,THIRDPARTY,PRE_CORE,DTYPES,FIRSTPARTY,POST_CORE,LOCALFOLDER
 
 known_first_party=pandas
-known_third_party=Cython,numpy,python-dateutil,pytz,pyarrow,pytest
```
Comment (datapythonista) [inline setup.cfg]: can you undo this change here, and send it in a different PR? I don't think this is related to the rest of the changes, is it?

File: setup.cfg [hunk 3]
```diff
@@ -45,19 +45,6 @@ ignore = E402,  # module level import not at top of file
...
-    doc/source/whatsnew/v0.9.0.rst
-    doc/source/whatsnew/v0.9.1.rst
-    doc/source/whatsnew/v0.10.0.rst
-    doc/source/whatsnew/v0.10.1.rst
-    doc/source/whatsnew/v0.11.0.rst
-    doc/source/whatsnew/v0.12.0.rst
-    doc/source/whatsnew/v0.13.0.rst
-    doc/source/whatsnew/v0.13.1.rst
-    doc/source/whatsnew/v0.14.0.rst
-    doc/source/whatsnew/v0.14.1.rst
```
Comment (datapythonista) [inline setup.cfg]: I'm a bit confused about this. Are all the others correct and we forgot to remove them? I'd expect just the `v0.13.*` to be removed

**Review by datapythonista — COMMENTED** [review]
Good changes, just added a few comments. Also, you'll have to fix the conflicts, I guess we removed some more fixed pages from `setup.cfg`

File: setup.cfg (hunk 3 above)
Comment (saurav2608) [reply setup.cfg]: Sorry if I am butting in. Issues related to v0.14* was fixed and merged. I think a merge with upstream/master will resolve this.

File: doc/source/whatsnew/v0.13.0.rst [hunk 4]
```diff
@@ -123,21 +119,30 @@ API changes
...
 
-      if df:
-         ....
-      df1 and df2
-      s1 and s2
+     ValueError: The truth value of a DataFrame is ambiguous.  Use a.empty,
+     a.bool(), a.item(), a.any() or a.all().
+
+     >>> df1 and df2
+     ValueError: The truth value of a DataFrame is ambiguous.  Use a.empty,
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.13.0.rst]: did you see if this generates a warning when building the docs? we may need to use this syntax if that's the case: https://docs.python.org/3/library/doctest.html (see `Traceback`...)

**Review by datapythonista — COMMENTED** [review]
Can you do the merge from master, and fix the conflicts please. We can't merge this PR otherwise.

**Comment by jreback** [comment]
pls merge master

File: doc/source/whatsnew/v0.13.0.rst (hunk 4 above)
Comment (Seuss27) [reply doc/source/whatsnew/v0.13.0.rst]: This throws an indention error and then an undefined error when the df1 and df2 gets indented (to aling with the if df: above I think.  Since df1, df2, s1, and s2 are never defined I wasn't sure how to proceed.  Should I define them or follow the traceback approach?

File: setup.cfg (hunk 3 above)
Comment (Seuss27) [reply setup.cfg]: Thanks for the feedback on this, I hadn't had to resolve a merge conflict before and I was doing it wrong.  All sorted now I believe.

File: setup.cfg (hunk 2 above)
Comment (Seuss27) [reply setup.cfg]: setup.cfg was rolled back completely and corrected.

File: doc/source/whatsnew/v0.13.0.rst (hunk 4 above)
Comment (Seuss27) [reply doc/source/whatsnew/v0.13.0.rst]: I ended up assigning the variables, let me know if you'd like to see an alternative solution...

**Comment by jreback** [comment]
thanks @Seuss27
